### PR TITLE
Clean interface

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | https://github.com/yiisoft/translator/issues/60

P.S. `demo` works fine without any changes

### CODE REVIEW

**Comment by xepozz** [comment]
Does anyone know what's the purpose of [this](https://github.com/yiisoft/translator/blob/860cf0f77698ae7f89c021ee70f36fbc398c0848/tests/TranslatorTest.php#L311) functionality?

File: README.md [hunk 1]
```diff
@@ -284,13 +265,6 @@ $newDefaultLocale = 'de-DE';
 echo $translator->withLocale($newDefaultLocale);
 ```
 
-### Get a new Translator instance with a category to be used by default in case category isn't specified explicitly.
-
-```php
-$newDefaultCategoryId = 'module2';
-echo $translator->withCategory($newDefaultCategoryId);
```
Comment (samdark) [inline README.md:291]: It wasn't removed, was it?

File: src/Translator.php [hunk 2]
```diff
@@ -33,14 +33,17 @@ final class Translator implements TranslatorInterface
     public function __construct(
         string $locale,
         ?string $fallbackLocale = null,
+        array $categorySources = [],
```
Comment (samdark) [inline src/Translator.php:36]: phpdoc missing.

**Comment by samdark** [comment]
Decided to leave that as is for now and decided on it later. Will add additional comments to the issue.

**Pull request closed without merging**